windows/tools/bitmap_converter_window.py

```diff
@@ -1277,7 +1277,7 @@ def make_log(self):
             formatted_strs[typ] = [''] * 18
             tag_info_strs[typ]  = [''] * 18
 
-            for fmt in range(len(BITMAP_FORMATS)):
+            for fmt in range(min(len(BITMAP_FORMATS), 18)):

```

Comment:
Idk what you're doing here. It's out of scope of this pr, but if you're going to do it, explain it.

Reply:
Due to supporting halo 2 and halo 3, I ended up implementing more bitmap formats than our 18. Because BITMAP_FORMATS is pulled from the globally defined halo bitmap formats enum, there will be more than 18 in it, thus causing an IndexError when it tries to access formatted_strs[typ][18]

A proper fix would be to define a constant at the top of the file for the max number of halo 1 formats, and work from that like so.
```python

HALO_1_TYPE_COUNT   = 4
HALO_1_FORMAT_COUNT = 18
for typ in range(HALO_1_TYPE_COUNT):
    formatted_strs[typ] = [''] * HALO_1_FORMAT_COUNT
    tag_info_strs[typ]  = [''] * HALO_1_FORMAT_COUNT

    for fmt in range(HALO_1_FORMAT_COUNT):
        if "?" not in BITMAP_FORMATS[fmt]:
            formatted_strs[typ][fmt] = "\n\n\t%s" % BITMAP_FORMATS[fmt]
            tag_info_strs[typ][fmt] = {}
```

I am fixing this now because it was preventing me from testing other changes, as it would always exception when making a log.